File: frontend/src/app/features/task/models/task.dto.ts
import {TaskStatusDto} from "./task-status.dto";
import {ProjectDto} from "../../project/models/project.dto";
import {UserDto} from "../../user/models/user.dto";
import {FileDto} from "../../../core/models/file.dto";
import {TaskCommentDto} from "./task-comment.dto";
import {TaskPriorityDto} from "./task-priority.dto";
import {ReplacedPartDto} from "./replaced-part.dto";

export interface TaskDto {
    taskId: number;
    projectId: number;
    project: ProjectDto;
    taskStatus: TaskStatusDto;
    taskPriority: TaskPriorityDto;
    taskName: string;
    taskDescription: string;
    taskDueDate: Date;
    taskEndDate: Date | null;
    taskRating: number;
    taskRatingComment: string;
    txDate: Date;
    txHost: string;
    taskAssignees: UserDto[];
    taskFiles: FileDto[];
    taskComments: TaskCommentDto[];
    replacedParts: ReplacedPartDto[];
}

File: backend/src/main/kotlin/bo/edu/umsa/backend/dto/TaskDto.kt
package bo.edu.umsa.backend.dto

import java.sql.Timestamp

data class TaskDto(
    val taskId: Int,
    val project: ProjectPartialDto,
    val taskStatus: TaskStatusDto,
    val taskPriority: TaskPriorityDto,
    val taskName: String,
    val taskDescription: String,
    val taskDueDate: Timestamp,
    val taskEndDate: Timestamp?,
    val taskRating: Int,
    val taskRatingComment: String,
    val txDate: Timestamp,
    val txHost: String,
    val taskAssignees: List<UserPartialDto>,
    val taskFiles: List<FilePartialDto>,
    val taskComments: List<TaskCommentDto>,
    val replacedParts: List<ReplacedPartDto>,
)

